task_list_app/task_list_app/src/main/java/com/jackson/task_list_app/api/controller/TaskController.java

```diff
@@ -1,51 +1,90 @@
 package com.jackson.task_list_app.api.controller;
 
+import java.util.Collections;
 import java.util.List;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.jackson.task_list_app.api.models.Task;
+import com.jackson.task_list_app.api.models.TaskStatus;
 import com.jackson.task_list_app.api.services.TaskService;
 
-
-
 @RestController
 public class TaskController {
 
     @Autowired
     private TaskService taskService;
 
+    // GetMappings
     @GetMapping("/tasks")
     @ResponseStatus(HttpStatus.OK)
-    public List<Task> getTask() {
-        return taskService.getTask();
+    public List<Task> getTasks(@RequestParam(required = false) String status) {
+        if (status != null) {
+            try {
+                TaskStatus taskStatus = TaskStatus.valueOf(status.toUpperCase());
+                return taskService.getTasksByStatus(taskStatus);
+            } catch (IllegalArgumentException e) {
+                // Handle invalid status value
+                return Collections.emptyList();
```

Comment:
Please return a `ResponseEntity` so you can override the `HTTP 200 Ok` status and send a `HTTP 400 Bad Request` or something. It would be odd to get a success code back when I supplied the status of "KJFKSJ."

Reply:
Got it, just changed it!